models/lenet/lenet.hpp

```diff
@@ -0,0 +1,202 @@
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   * @param padW Padding width of the input.
+   * @param padH Padding height of the input.
+   */
+  void ConvolutionBlock(const size_t inSize,
+                        const size_t outSize,
+                        const size_t kernelWidth,
+                        const size_t kernelHeight,
+                        const size_t strideWidth = 1,
+                        const size_t strideHeight = 1,
+                        const size_t padW = 0,
+                        const size_t padH = 0)
+  {
+    leNet->Add<Convolution<>>(inSize, outSize, kernelWidth,
+        kernelHeight, strideWidth, strideHeight, padW, padH, inputWidth,
+        inputHeight);
+    leNet->Add<LeakyReLU<>>();
+
+    // Update inputWidth and input Height.
+    inputWidth = ConvOutSize(inputWidth, kernelWidth, strideWidth, padW);
+    inputHeight = ConvOutSize(inputHeight, kernelHeight, strideHeight, padH);
+    return;
+   }
+
+  /**
+   * Adds Pooling Block.
+   *
+   * @param kernelWidth Width of the filter/kernel.
+   * @param kernelHeight Height of the filter/kernel.
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   */
+  void PoolingBlock(const size_t kernelWidth,
+                    const size_t kernelHeight,
+                    const size_t strideWidth = 1,
+                    const size_t strideHeight = 1)
+  {
+    leNet->Add<MaxPooling<>>(kernelWidth, kernelHeight,
+        strideWidth, strideHeight, true);
+    // Update inputWidth and inputHeight.
+    inputWidth = PoolOutSize(inputWidth, kernelWidth, strideWidth);
+    inputHeight = PoolOutSize(inputHeight, kernelHeight, strideHeight);
+    return;
+  }
+
+  /**
+   * Return the convolution output size.
+   *
+   * @param size The size of the input (row or column).
+   * @param k The size of the filter (width or height).
+   * @param s The stride size (x or y direction).
+   * @param padding The size of the padding (width or height) on one side.
+   * @return The convolution output size.
+   */
+  size_t ConvOutSize(const size_t size,
+                     const size_t k,
+                     const size_t s,
+                     const size_t padding)
+  {
+    return std::floor(size + 2 * padding - k) / s + 1;
+  }
+
+  /**
+   * Return the convolution output size.
+   *
+   * @param size The size of the input (row or column).
+   * @param k The size of the filter (width or height).
+   * @param s The stride size (x or y direction).
+   * @return The convolution output size.
+   */
+  size_t PoolOutSize(const size_t size,
+                     const size_t k,
+                     const size_t s)
+  {
+    return std::floor(size - 1) / s + 1;
+  }
+
+  //! Locally stored LeNet Model.
+  Sequential<> *leNet;
```

Comment:
```suggestion
  Sequential<>* leNet;
```
Also, maybe it would be better to store `leNet` as an `FFN` instead?  (If one wanted to use `LeNet` inside of other models, it would be easy to wrap it in a `Sequential<>` then.  But at least `object_recognition.cpp` uses an FFN that's only `LeNet`, so that code could be simplified with `leNet` as just an `FFN`. :+1: